api/src/tasks/generateNbTxsPerDay.ts

```diff
@@ -1,29 +1,36 @@
 import { addDays, isBefore, format } from "date-fns";
 import { prisma } from "../prisma";
-import { startDate } from "../utils";
+import { startDate, Result } from "../utils";
 
-export async function generateNbTxsPerDay() {
+export async function generateNbTxsPerDay(currentData: Result[]) {
   const endDate = new Date();
   let iteratorDate = startDate;
```

Comment:
@Akirtovskis sipmle solution could be to set the iteratorDate value to the latest populated date coming from `currentData`.